packages/flutter_test/lib/src/matchers.dart

```diff
@@ -277,6 +278,144 @@ Matcher matchesGoldenFile(dynamic key) {
   throw new ArgumentError('Unexpected type for golden file: ${key.runtimeType}');
 }
 
+/// Asserts that a [SemanticsData] contains the specified information.
+///
+/// If the either the label, hint, value, textDirection, or rect fields are not
+/// provided, then they are not part of the comparison.  All of the boolean
+/// flag and action fields must match, and default to false.
+///
+/// To retrieve the semantics data of a widget, use [tester.getSemanticsData]
+/// with a [Finder] that returns a single widget.  Semantics must be enabled
+/// in order to use this method.
+///
+/// ## Sample code
+///
+/// ```dart
+/// final SemanticsHandle handle = tester.ensureSemantics();
+/// final SemanticsData data = tester.getSemanticsData(find.text('hello'));
+/// expect(data, matchesSemanticsData(label: 'hello'));
+/// handle.dispose();
+/// ```
+///
+/// See also:
+///
+///   * [WidgetTester.getSemanticsData], the tester method which retrieves data.
+Matcher matchesSemanticsData({
+  String label,
+  String hint,
+  String value,
+  TextDirection textDirection,
+  Rect rect,
+  // Flags //
+  bool hasCheckedState = false,
+  bool isChecked = false,
+  bool isSelected = false,
+  bool isButton = false,
+  bool isFocused = false,
+  bool isTextField = false,
+  bool hasEnabledState = false,
+  bool isEnabled = false,
+  bool isInMutuallyExclusiveGroup = false,
+  bool isHeader = false,
+  bool isObscured = false,
+  bool namesRoute = false,
+  bool scopesRoute = false,
+  bool isHidden = false,
+  // Actions //
+  bool hasTap = false,
```

Comment:
Should we be more specific here and name these hasTapAction, hasLongPressAction? Or is that too verbose?

Reply:
Done